### DISCUSSION TITLE
Is there a way to keep context in tags with mini.surround?

### DISCUSSION THREAD

**Discussion by nandcello** [nvim-mini/mini.nvim#1375] [Q&A] [answered]
When working with html tags, by context I mean attributes
```html
<div class="cool" id="stuff">
...
</div>
```

So in https://github.com/tpope/vim-surround for example I have two options to replace the div tag which is to replace it entirely with `cst<` then the tag name and closing bracket (example `span>` which produces this result
```
<span>
...
</span>
```

The second option is keeping the class and id attributes which can be done with `cst<` as well but then when you do only the tag name `span` it will keep the attributes so produces the following result
```
<span class="cool" id="stuff">
...
</span>
```

**Comment by echasnovski** [comment, marked as answer]
The `t` (tag) surrounding will always delete/replace the whole tag with all attributes inside. The approach here for adjusting only tag name is to create a separate surrounding specifically for this. For example, under `T` surrounding as [described here](https://github.com/echasnovski/mini.nvim/issues/1293#issuecomment-2423827325). With this, you can `srTT` to replace tag name with another tag name.

**Reply by nandcello** [reply]
Thanks for the help and making such a great plugin!
